### PR TITLE
Added basic Login screen with mock vdk function call

### PR DESCRIPTION
**Pull request by kressnick25**
Gets a result from function call on submit, handles errors and displays message if necessary.

### CODE REVIEW

File: lib/auth/login.dart [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+import 'package:flutter/material.dart';
+
+import '../util/Result.dart';
+
+class AuthDetails {
+  String username;
+  String password;
+
+  AuthDetails({this.username, this.password});
+}
```
Comment (yuk0ga) [inline lib/auth/login.dart:11]: This class might need to change later for better security? I wonder what the best practice is for handling user credentials in flutter..

Comment (yuk0ga) [reply lib/auth/login.dart:11]: https://github.com/Euclideon/vaultclient/blob/master/src/vcSession.cpp#L135
This is probably where they handle login in VaultClient but not quite sure how it works haha
But might be worth exploring

Comment (kressnick25) [reply lib/auth/login.dart:11]: In their example the password gets passed in from the program state object so this is equivalent to that I guess. After the password gets passed to the vdkContext object we won't need the password anymore so can discard after that.

Comment (jmkraut) [reply lib/auth/login.dart:11]: This is what I was wondering, if we are making a call to a database, will we have to hash and salt these things or is that not a matter of importance at this point? Might be easier to work with dummy data for now without worrying about it too much.

Comment (kressnick25) [reply lib/auth/login.dart:11]: The vdk handles the network request so we rely on that handling security properly. Here the value of the password is only stored in memory not saved to disk anywhere. If someone can attack the memory then we have much bigger problems. See here https://security.stackexchange.com/questions/29019/are-passwords-stored-in-memory-safe. Also we don't intend to keep this password in application global state, this class is just to give some shape to the ephemeral form state. It's just used to create the vdkContext object, after that we don't need the password

**Pull request merged**